pkg/operator/metricscontroller/legacy_cn_certs_sync.go

```diff
@@ -30,7 +31,11 @@ func NewLegacyCNCertsMetricsSyncFunc(query string, operatorClient v1helpers.Oper
 		var value model.SampleValue
 		switch result.Type() {
 		case model.ValVector:
-			value = result.(model.Vector)[0].Value
+			vec := result.(model.Vector)
+			if len(vec) == 0 {
+				return errors.New("query returned empty vector")
```

Comment:
Why is this an error? Why can't the vector just be empty?

You should use `fmt.Errorf` instead of `errors.New` as in the rest of the code

Reply:
It is an error as we have no data available and cannot interpret it correctly. Similar to here https://github.com/openshift/cluster-etcd-operator/blob/master/pkg/operator/metriccontroller/fsync_controller.go#L64 there is little we can do.

Here, concretely, it is a transient error. This controller doesn't set a degraded status but rather retries during the next sync.

Reply:
why should we use `fmt.Errorf` here? there is no parameter to be set.

Reply:
I think returning an error is the correct behavior here since an empty result is unexpected and most likely means that no timeserie like the following one exist in Prometheus yet:
```
apiserver_webhooks_x509_missing_san_total{job="apiserver",namespace="default"}
```
That can happen for various reasons unrelated to this particular controller, so in the case where it isn't able to execute the query or get a proper result, I would consider that an error since the controller can't proceed with providing a correct status.

Reply:
I see, I did not have this kind of understanding, in my mind this was just that the metrics was uninitialized which would mean that having the `value` be 0 would be correct in my mind.